### PR TITLE
Add mainnet config

### LINKED ISSUE #503: add mainnet config
lol

### PR DESCRIPTION
**Pull request by ninjaahhh**
double check, no, triple check before merging this

recommend using diff tool to compare with testnet2.5

will fix #503

### CODE REVIEW

File: mainnet/singularity/cluster_config_template.json [hunk 1]
```diff
@@ -0,0 +1,419 @@
...
+          "GAS_LIMIT": 12000000,
+          "NONCE": 0,
+          "ALLOC": {
+            "d50f23e410711c1391f5ef88fc11245e564c7684000775b2": {
+              "QKC": 750000000000000000000000000
+            }
+          }
+        },
+        "COINBASE_ADDRESS": "000000000000000000000000000000000000000000000000",
+        "COINBASE_AMOUNT": 3250000000000000000,
```
Comment (qizhou) [inline mainnet/singularity/cluster_config_template.json]: Coinbase should be 6.5 QKC (pre tax) where 3.25 QKC is post tax

File: mainnet/singularity/cluster_config_template.json [hunk 2]
```diff
@@ -0,0 +1,419 @@
...
+    "GUARDIAN_PRIVATE_KEY": null,
+    "P2P_PROTOCOL_VERSION": 0,
+    "P2P_COMMAND_SIZE_LIMIT": 134217728,
+    "SKIP_ROOT_DIFFICULTY_CHECK": false,
+    "SKIP_MINOR_DIFFICULTY_CHECK": false,
+    "GENESIS_TOKEN": "QKC",
+    "ENABLE_TX_TIMESTAMP": 1561791600,
+    "ENABLE_EVM_TIMESTAMP": 1569567600,
+    "ROOT": {
+      "MAX_STALE_ROOT_BLOCK_HEIGHT_DIFF": 60,
```
Comment (qizhou) [inline mainnet/singularity/cluster_config_template.json]: Should be 22500 according to geth and our latest config (config.py)

File: quarkchain/tools/config_slave.py [hunk 3]
```diff
@@ -10,6 +10,8 @@
 import os
 
 FILE = "../../testnet/2/cluster_config_template.json"
+if "config" in os.environ:
```
Comment (qizhou) [inline quarkchain/tools/config_slave.py]: Probably use upper case with more info such as QKC_CONFIG

**Comment by ninjaahhh** [comment]
@qizhou be careful when you force push

**Comment by ninjaahhh** [comment]
proposed launching schedule, suppose the target time for launching the network is `t`:

0. `t - 48hr`. announce `t` to the public. note that we should clarify it's not exact time and allows some leeway (say, ±1hr)
1. `t - 24hr`. finalize following parts of the mainnet config and keep it private
    - genesis timestamp and data to avoid pre-mining
    - configuration parameters like posw config and tx sender whitelist
    - reserved IP for boot nodes. note that the machines won't be started yet
    - p2p priv/pub key for aforementioned boot nodes
2. `t - 2hr`: launch the mainnet explorer, but it doesn't have any network to connect to
3. around `t`
    - launch the boot nodes
    - launch the indexing node
    - launch the guardian node
    - release the mainnet config by pushing to the repo's master branch
    - issue public announcements

**Comment by qizhou** [comment]
I guess we for t - 2 hours, all of the nodes should be started with the mainnet config, but the config and release will be done at t?

**Pull request merged**